### PR TITLE
build: manually update PyTorch version

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Set PyTorch and TorchVision version to nightly release 2024-04-01.

Signed-Off By: Vivek Khandelwal <vivekkhandelwal1424@gmail.com>

### CODE REVIEW

**Comment by vivekkhandelwal1** [comment]
@stellaraccident who's the right person to ask for looking at the LTC failures?

**Comment by stellaraccident** [comment]
You can ask on discord, but the agreement is to disable and keep moving.

**Review by stellaraccident — APPROVED** [review]

File: lib/Dialect/Torch/Transforms/AbstractInterpLibrary.cpp [hunk 1]
```diff
@@ -5868,6 +5868,24 @@ StringRef mlir::torch::Torch::getAbstractInterpLibrary() {
 "    %3 = torch.prim.TupleConstruct %1, %0, %0 : !torch.list<int>, !torch.list<int>, !torch.list<int> -> !torch.tuple<list<int>, list<int>, list<int>>\n"
 "    return %3 : !torch.tuple<list<int>, list<int>, list<int>>\n"
 "  }\n"
+"  func.func @__torch__.torch.jit._shape_functions._batch_norm_with_update(%arg0: !torch.list<int>, %arg1: !torch.optional<list<int>>, %arg2: !torch.optional<list<int>>, %arg3: !torch.optional<list<int>>, %arg4: !torch.optional<list<int>>) -> !torch.tuple<list<int>, list<int>, list<int>, list<int>> {\n"
```
Comment (qingyunqu) [inline lib/Dialect/Torch/Transforms/AbstractInterpLibrary.cpp:5871]: I think that you forget to adding shape function of `_batch_norm_with_update` in `abstract_interp_lib_gen.py`? @vivekkhandelwal1

Comment (qingyunqu) [reply lib/Dialect/Torch/Transforms/AbstractInterpLibrary.cpp:5871]: But why the ci passed?

Comment (vivekkhandelwal1) [reply lib/Dialect/Torch/Transforms/AbstractInterpLibrary.cpp:5871]: > I think that you forget to adding shape function of `_batch_norm_with_update` in `abstract_interp_lib_gen.py`? @vivekkhandelwal1

No, this change comes from the upstream addition of this shape function.

Comment (vivekkhandelwal1) [reply lib/Dialect/Torch/Transforms/AbstractInterpLibrary.cpp:5871]: > But why the ci passed?

The same reason as above + we disabled the LTC build.

**Pull request merged**